technical-reports/resolver/resolution-logic.md

```diff
@@ -21,48 +18,35 @@ If a resolver does NOT declare any modifiers, skip this step and proceed to [Set
 2. For every modifier in the resolver:
    1. If that resolver does NOT declare a default value, verify a key is provided in the input. If not, throw an error.
 
-## Set flattening
+## Token flattening
 
-Tools MUST iterate over the resolver’s [sets syntax](#sets) in order.
+Tools MUST iterate over the [tokens](#tokens) array in order.
 
-1. Starting with the first set:
-   1. Load the first item in the `values` array to form the **basis** for all tokens.
-   2. Load the next item in the `values` array, merging the objects together.
-   3. Resolve conflicts according to [conflict resolution](#conflict-resolution).
-   4. [Aliases](../format/#aliases-references) MUST NOT be resolved yet. That must happen at the end. Aliases MAY refer to values that will be supplied in upcoming steps.
-   5. Continue loading the next item in the `values` array, repeating steps 2–3.
-   6. After all `values` have been merged into the **basis** object, keep that in memory and continue onto the next set.
-2. Continue onto the set, loading tokens in the same way as before.
-   1. Repeating the steps previously, you’ll also end up with a **basis** for this set—a single object containing all tokens referenced in the set.
-3. Continue one-by-one through the remaining sets, until you have a collection of one **basis** object per set.
-4. Return back to the first set, first basis, then merge the second set, second basis, and so on, until you reach the final set.
-   1. [Resolve conflicts](#conflict-resolution) as every additional set is merged.
-5. After all sets have been merged, there will be one single tokens object containing all tokens referenced.
+### Sets
 
-### Conflict resolution
-
-Conflict resolution occurs when flattening [=sets=] or applying [=modifiers=], and a token name is occupied by tokens of different values, from different sources. In many cases, this is intentional, but not always.
-
-When 2 tokens try and occupy the same space, tools MUST resolve the conflict in the following manner:
-
-1. If the token types are **identical**, overwrite the latter value with the former.
-1. If the token types are **incompatible**, the tool MUST throw an error.
-1. If one namespace is a token, and the other is a group, the tool MUST throw an error.
-1. If one value is an alias (i.e. the `$type` is unknown), overwrite the value.
+1. Sets are loaded in the [tokens](#tokens) array order.
+1. If this uses [simple syntax](#simple-syntax) (string), treat it as if it is a set with no `name`, with a single item in the `sources` array.
+1. Every token reference in `sources` MUST be resolved in array order.
+1. For every token reference, merge with the previous set.
+1. [Aliases](../format/#aliases-references) MUST NOT be resolved yet. That must happen at the end. Aliases MAY refer to values that will be supplied in upcoming steps.
 
 <aside class="example" title="Conflict resolution">
 
 ```json
 {
-  "sets": [
+  "tokens": [
```

Comment:
`"tokens"` is a placeholder, we’re not really crazy about and is subject to change. We just need “something” here and anything will suffice.